src/messaging/messaging.ios.ts

```diff
@@ -516,27 +517,22 @@ function _addObserver(eventName, callback) {
   return NSNotificationCenter.defaultCenter.addObserverForNameObjectQueueUsingBlock(eventName, null, NSOperationQueue.mainQueue, callback);
 }
 
-// see https://developer.apple.com/reference/usernotifications/unusernotificationcenterdelegate?language=objc
-class UNUserNotificationCenterDelegateImpl extends NSObject implements UNUserNotificationCenterDelegate {
-  public static ObjCProtocols = [];
-
-  static new(): UNUserNotificationCenterDelegateImpl {
-    if (UNUserNotificationCenterDelegateImpl.ObjCProtocols.length === 0 && typeof (UNUserNotificationCenterDelegate) !== "undefined") {
-      UNUserNotificationCenterDelegateImpl.ObjCProtocols.push(UNUserNotificationCenterDelegate);
-    }
-    return <UNUserNotificationCenterDelegateImpl>super.new();
-  }
+class FirebaseNotificationDelegateObserverImpl implements DelegateObserver {
+  observerUniqueKey = "firebase-messaging";
 
   private callback: (unnotification: UNNotification, actionIdentifier?: string, inputText?: string) => void;
 
-  public initWithCallback(callback: (unnotification: UNNotification, actionIdentifier?: string, inputText?: string) => void): UNUserNotificationCenterDelegateImpl {
+  constructor(callback: (unnotification: UNNotification, actionIdentifier?: string, inputText?: string) => void) {
     this.callback = callback;
-    return this;
   }
 
-  public userNotificationCenterWillPresentNotificationWithCompletionHandler(center: UNUserNotificationCenter, notification: UNNotification, completionHandler: (p1: UNNotificationPresentationOptions) => void): void {
+  public userNotificationCenterWillPresentNotificationWithCompletionHandler(center: UNUserNotificationCenter, notification: UNNotification, completionHandler: (p1: UNNotificationPresentationOptions) => void, next: () => void): void {
     const userInfo = notification.request.content.userInfo;
     const userInfoJSON = firebaseUtils.toJsObject(userInfo);
+    if (!userInfoJSON["gcm.message_id"]) { // not a firebase message!
+      next();
+      return;
+    }
 
```

Comment:
This check should also be changed similar to the one below so it can work with non-FCM pushes.